src/Randomizer.SMZ3.Tracking/TrackerServiceCollectionExtensions.cs

```diff
@@ -33,18 +33,18 @@ public static IServiceCollection AddTracker<TWorldAccessor>(this IServiceCollect
             services.AddScoped<TrackerModuleFactory>();
 
             services.AddSingleton<IWorldAccessor>(x => x.GetRequiredService<TWorldAccessor>());
-            services.AddSingleton<TrackerConfigProvider>();
-            services.AddSingleton(serviceProvider =>
+            services.AddScoped<TrackerConfigProvider>();
```

Comment:
The provider itself can stay singleton, as it doesn't cache the instances it creates. Not too important unless we need it somewhere that doesn't use a service scope.

Reply:
Updated